app/javascript/components/utils/AvailabilityTimeFormatPrefs.js

```diff
@@ -0,0 +1,27 @@
+
+const TIME_FORMAT_PREF = 'timeFormatPreference';
+const TWELVE_HOUR_FORMAT = '12-hour';
+const TWENTY_FOUR_HOUR_FORMAT = '24-hour';
+
+class AvailabilityTimeFormatPrefs {
+  static set12HourPref() {
+    localStorage.setItem(TIME_FORMAT_PREF,TWELVE_HOUR_FORMAT); 
+  }
+  static set24HourPref() {
+    localStorage.setItem(TIME_FORMAT_PREF,TWENTY_FOUR_HOUR_FORMAT); 
+  }
+
+  static use12HourFormat() {
+    return !(localStorage.getItem(TIME_FORMAT_PREF) === TWENTY_FOUR_HOUR_FORMAT);
+  }    
+
+  static toggleTimePref(using12Hour) {
+    if(using12Hour) {
+      AvailabilityTimeFormatPrefs.set24HourPref();
+    }else {
+      AvailabilityTimeFormatPrefs.set12HourPref();
+    }
+  }
+}
```

Comment:
I would recommend only providing a 12 hour formatting option. The idea of abstracting this out and the way you executed it looks great, feels like a shame to cut it, but that's just my two cents.

Reply:
Thanks! I didn't want to get rid of the functionality for users to be able to see the time in a 24-hour format. Since outside of the US, the 24-hour notation is common. I thought it might be more user friendly to give these users an option to view the time in the format more familiar to them.